File: 0001-two-sum/hash-map/hash-map.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {};

        for value, key in enumerate(nums):
            required = target - key;

            # Check if required is in dictionary
            if required in hashMap:
                return [hashMap[required], value];
            else:
                hashMap[key] = value;
        
        return [];
        

File: 0001-two-sum/hash-map/hash-map.cs
using System.Collections.Generic;

public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        // Create dictionary to act as our hash map
        Dictionary<int, int> hashMap = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++)
        {
            int required = target - nums[i];

            // Check the map
            if (hashMap.ContainsKey(required))
            {
                return new int[] { hashMap[required], i};
            }
            else
            {
                // Otherwise, store the key in the map
                hashMap[nums[i]] = i;
            }
        }

        return new int[0];
    }
}